include/bouncer.h

```diff
@@ -76,9 +76,28 @@ enum PauseMode {
 };
 
 enum ShutDownMode {
+	/* just stay running */
 	SHUTDOWN_NONE = 0,
-	/* wait for all servers to become idle before stopping the process */
+	/*
+	 * Wait for all servers to become idle before stopping the process. New
+	 * clients will still accepted, but they are not given connections from
+	 * the pool. Already connected clients which are unlinked are also not
+	 * given servers from the pool again.
+	 *
+	 * This is effectively the same as running PAUSE from the admin console
+	 * and then, once all clients are paused, you run SHUTDOWN (which
+	 * translates to SHUTDOWN_IMMEDIATE).
+	 */
 	SHUTDOWN_WAIT_FOR_SERVERS,
+	/*
+	 * Wait for all clients to disconnect before stopping the process. Stop
+	 * listening on the socket so no new clients can connect. Still
+	 * connected clients will continue to be handed connections from the
+	 * pool until they disconnect.
+	 *
+	 * This allows for a rolling restart in combination with so_reuseport.
+	 */
+	SHUTDOWN_WAIT_FOR_CLIENTS,
```

Comment:
I suppose "wait for clients" kind of includes "wait for servers"? Not sure if that is worth writing down, just thinking out loud.

Reply:
> I suppose "wait for clients" kind of includes "wait for servers"?

This wasn't the case before, but with my most recent changes it does. I updated the comment to reflect this.